src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
```

Comment:
I was using the term "rebase" loosely, but I am actually not convinced either way as far as rebasing vs merge. It is definitely a discussion worth having when we get to that point. I can see arguments for both sides.

> we would lose the ability to track when a change was made.

What do you mean? I have already done an interactive rebase on vscode/patch once to squash some commits together (to reduce cumulative diff) and it appears to have maintained the original author and commit date.

<img width="508" alt="screen shot 2017-02-26 at 3 40 50 pm" src="https://cloud.githubusercontent.com/assets/754768/23344933/fe0ddfb4-fc39-11e6-8e6a-b1b33dfb8b98.png">